datadog/fwprovider/resource_datadog_apm_retention_filter_order.go

```diff
@@ -0,0 +1,187 @@
+	r.Auth = providerData.Auth
+}
+
+func (r *ApmRetentionFiltersOrderResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "apm_retention_filter_order"
+}
+func (d *ApmRetentionFiltersOrderResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Provides a Datadog [APM Retention Filters API](https://docs.datadoghq.com/api/v2/apm-retention-filters/) resource, which is used to manage Datadog APM retention filters order.",
+		Attributes: map[string]schema.Attribute{
+			"filter_ids": schema.ListAttribute{
+				Description: "The filter IDs list. The order of filters IDs in this attribute defines the overall APM retention filters order. If `filter_ids` is empty or not specified, it will import the actual order, and create the resource. Otherwise, it will try to update the order.",
+				ElementType: types.StringType,
+				Required:    true,
+			},
+			"id": utils.ResourceIDAttribute(),
+		}}
+}
+
+func (r *ApmRetentionFiltersOrderResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *ApmRetentionFiltersOrderResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state ApmRetentionFiltersOrderModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resp, httpResp, err := r.Api.ListApmRetentionFilters(r.Auth)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error retrieving retention filter"))
+		return
+	}
+	if err := utils.CheckForUnparsed(resp); err != nil {
+		response.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+
+	r.updateState(ctx, &state, &resp)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *ApmRetentionFiltersOrderResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var state ApmRetentionFiltersOrderModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if len(state.FilterIds) > 0 {
+		body, diags := r.buildRetentionFiltersOrderRequestBody(ctx, &state)
+		response.Diagnostics.Append(diags...)
+		if response.Diagnostics.HasError() {
+			return
+		}
+		_, err := r.Api.ReorderApmRetentionFilters(r.Auth, *body)
+		if err != nil {
+			response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error re-ordering retention filters"))
+			return
+
+		}
+	}
+	listData, httpResponse, err := r.Api.ListApmRetentionFilters(r.Auth)
```

Comment:
What do you think about removing this call and setting what user provided directly to state? I have not looked into the backend implementation but is there a potential of replication delay between `ReorderApmRetentionFilters` call and what is returned by `ListApmRetentionFilters`? This can cause inconsistent state failures. Same with the call in `func Update`

Reply:
my idea was that we should always prioritize the data in the API, so even if the request fail, we set the state from API data, so the next run we will be able to detect the issue in terraform before we even send a call to the API.
There shouldn't be any replication delay between the calls.